app/models/manageiq/providers/ansible_tower/automation_manager/machine_credential.rb

```diff
@@ -1,2 +1,66 @@
 class ManageIQ::Providers::AnsibleTower::AutomationManager::MachineCredential < ManageIQ::Providers::AutomationManager::Authentication
+  FIELDS = {
+    :description => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('Description'),
+      :help_text  => _('Optional description of this credential.'),
+      :encrypted  => false
+    },
+    :username         => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('Username'),
+      :help_text  => _('Username for this credential.'),
+      :max_length => 1024,
+      :encrypted  => false
+    },
+    :ssh_key_data     => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('SSH private key'),
+      :help_text  => _('RSA or DSA private key to be used instead of password.'),
+      :encrypted  => true
+    },
+    :ssh_key_unlock   => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('SSH key unlock'),
+      :help_text  => _('Passphrase to unlock SSH private key if encrypted (or \'ASK\' to prompt the user for machine credentials).'),
+      :max_length => 1024,
+      :encrypted  => true
+    },
+    :become_method    => {
+      :type       => :choice,
+      :required   => false,
+      :label      => _('Become method'),
+      :help_text  => _('Privilege escalation method.'),
+      :encrypted  => false,
+      :choices    => %w(sudo su pbrun pfexec).insert(0, '')
+    },
+    :become_username  => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('Become username'),
+      :help_text  => _('Privilege escalation username.'),
+      :max_length => 1024,
+      :encrypted  => false
+    },
+    :become_password => {
+      :type       => :string,
+      :required   => false,
+      :label      => _('Become password'),
+      :help_text  => _('Password for privilege escalation method.'),
+      :max_length => 1024,
+      :encrypted  => true
+    },
```

Comment:
Just to be clear, I think we should present extra_attributes for the UI, but I don't think we should *store* passwords in the extra attributes at all, otherwise we have to update fixauth, logging, etc.